npe2/_dynamic_plugin.py

```diff
@@ -206,7 +205,11 @@ def _store_command(self, func: T, kwargs: dict) -> dict:
         """Create a new command contribution for `func`"""
         kwargs.setdefault("title", func.__name__)
         kwargs.setdefault("id", f"{self.plugin.manifest.name}.{func.__name__}")
-        cmd_kwargs = {k: kwargs.pop(k) for k in list(kwargs) if k in COMMAND_PARAMS}
+        cmd_kwargs = {
+            k: kwargs.pop(k)
+            for k in list(kwargs)
+            if k in CommandContribution.__fields__
+        }
```

Comment:
```suggestion
	kwargs_set = set(kwargs)
        cmd_kwargs = {
            k: kwargs.pop(k)
            for k in kwargs_set
            if k in CommandContribution.__fields__
        }
```
I know that this is only a few positions, but creating a list every time? Some less experienced users may use this as inspiration in places where this condition is not satisfied.

Reply:
I mean you don't even need the set/list,  as 'k in kwargs' will iterate over the dict keys, and if you use the set, then why not `kwargs_set = set(kwargs).union(set(CommandContribution.__fields__))` and avoid the test.

Reply:
not union but intersection

Reply:
ahh, maybe to early. I replace in the head if check with iteration

Reply:
> ahh, maybe to early. I replace in the head if check with iteration

Ghhhhaaaa. Pfrrlrlrlflfr. Yes.

Reply:
... How many programmers do you need to get a dict comprehension right ...
Today is not a functioning brain day.

Reply:
> I mean you don't even need the set/list, as 'k in kwargs' will iterate over the dict keys,

we do in this case, since kwargs is getting changed on each iteration by `kwargs.pop`...

> but creating a list every time?

it's not creating a list on each iteration either way (the list would only get called on the first evaluation of the comprehension)

<details>

```python
In [13]: class T:
    ...:     def __iter__(self):
    ...:         print('iter')
    ...:         yield from ['a','b','c']
    ...:

In [14]: kwargs = T()

In [15]: list(kwargs)
iter
Out[15]: ['a', 'b', 'c']

In [16]: {k for k in list(kwargs) if k in ['a','b']}
iter
Out[16]: {'a', 'b'}

In [17]: {k for k in kwargs if k in ['a','b']}
iter
Out[17]: {'a', 'b'}
```

</details>